web/components/FcDrawerRequestStudyBulkEdit.vue

```diff
@@ -51,13 +61,26 @@ export default {
   beforeDestroy() {
     this.setLocationsIndicesDeselected([]);
   },
+  beforeRouteLeave(to, from, next) {
```

Comment:
make a new Vue mixin (e.g. `FcMixinRequestStudyLeaveGuard`) with these methods in it, and use it in all three places - this helps ensure consistency.